packages/network-controller/src/NetworkController.ts

```diff
@@ -160,9 +186,23 @@ export type NetworkState = {
 
 const name = 'NetworkController';
 
-export type BlockTrackerProxy = SwappableProxy<BlockTracker>;
+/**
+ * Represents the block tracker for the currently selected network. (Yes, this
+ * is a proxy around a proxy: the inner one exists to manage a network client
+ * that exists only after the block tracker is used, and the outer one exists
```

Comment:
Sure. What about:

> Note that this is a proxy around a proxy: the inner one exists so that the block tracker doesn't have to exist until it's used, and the outer one exists so that the currently selected network can change without consumers needing to refresh the object reference to that network.

Reply:
Sounds great!

Reply:
Updated here: Updated here: